server/config/logger.go

```diff
@@ -212,3 +212,87 @@ func makeFileOptions(filename string) (json.RawMessage, error) {
 
 	return json.RawMessage(b), nil
 }
+// makeAuditFileOptions builds file options for AUDIT logs using ExperimentalAuditSettings.
+// Falls back to the regular logging defaults when fields are unset.
+func makeAuditFileOptions(a *model.ExperimentalAuditSettings) (json.RawMessage, error) {
+	filename := ""
+	maxSize := LogRotateSizeMB
+	maxAge := LogRotateMaxAge
+	maxBackups := LogRotateMaxBackups
+	compress := LogCompress
+
+	if a != nil {
+		// NOTE: The field in ExperimentalAuditSettings is named FileName (not Filename). This is intentional; do not change it.
+		if a.FileName != nil {
+			filename = *a.FileName
+		}
+		if a.FileMaxSizeMB != nil {
+			maxSize = *a.FileMaxSizeMB
+		}
+		if a.FileMaxAgeDays != nil {
+			maxAge = *a.FileMaxAgeDays
+		}
+		if a.FileMaxBackups != nil {
+			maxBackups = *a.FileMaxBackups
+		}
+		if a.FileCompress != nil {
+			compress = *a.FileCompress
+		}
+	}
+
+	opts := struct {
+		Filename    string `json:"filename"`
+		Max_size    int    `json:"max_size"`
+		Max_age     int    `json:"max_age"`
+		Max_backups int    `json:"max_backups"`
+		Compress    bool   `json:"compress"`
+	}{
+		Filename:    filename,
+		Max_size:    maxSize,
+		Max_age:     maxAge,
+		Max_backups: maxBackups,
+		Compress:    compress,
+	}
```

Comment:
I wonder if we can combine this code with https://github.com/mattermost/mattermost/blob/852392e6c2496966141af1f8831b52f24e913528/server/config/logger.go#L193-L215